### PR TITLE
feature : today-screen

### PR DESCRIPTION
**Pull request by FabrizioFarfan**

### CODE REVIEW

File: app/src/main/java/co/develhope/meteoapp/MainActivity.kt [hunk 1]
```diff
@@ -16,6 +18,26 @@ class MainActivity : AppCompatActivity() {
...
+            TodayInfo("20:00",R.drawable.sun_image,"34°","13%","39°","5/10","15%","SSE 7km/h","30%","0cm"),
+            TodayInfo("21:00",R.drawable.sun_image,"33°","17%","28°","5/10","16%","SSE 7km/h","30%","0cm"),
+            TodayInfo("22:00",R.drawable.sun_image,"37°","19%","68°","5/10","17%","SSE 7km/h","30%","0cm"),
+            TodayInfo("23:00",R.drawable.sun_image,"36°","10%","47°","5/10","18%","SSE 7km/h","30%","0cm"),
+            TodayInfo("00:00",R.drawable.sun_image,"35°","18%","45°","5/10","20%","SSE 7km/h","30%","0cm")
+
+        )
+        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = TodayInfoAdapter(todayInfoList)
```
Comment (ShAdOwNeT88) [inline app/src/main/java/co/develhope/meteoapp/MainActivity.kt]: I don't think that this part is belong to MainActivity

File: app/src/main/java/co/develhope/meteoapp/TodayInfo.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    val temperature: String,
+    val humidity: String,
+    val percepita: String,
+    val indice: String,
+    val umidita: String,
+    val vento: String,
+    val copertura: String,
+    val pioggia: String,
+    var isExpanded: Boolean = false
+)
```
Comment (ShAdOwNeT88) [inline app/src/main/java/co/develhope/meteoapp/TodayInfo.kt]: TodayInfo maybe is too specific, let's rename this model as WeatherConditions, this way we can reuse it.

File: app/src/main/java/co/develhope/meteoapp/TodayInfoFragment.kt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package co.develhope.meteoapp
+
+class TodayInfoFragment {
+}
```
Comment (ShAdOwNeT88) [inline app/src/main/java/co/develhope/meteoapp/TodayInfoFragment.kt]: Why this could be empty?

File: app/src/main/res/layout/card_view_item.xml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+            <TextView
+                android:id="@+id/percepita"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/percepita"
+                android:textAppearance="@style/percepita" />
+
+            <TextView
+                android:id="@+id/some_id4"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/card_view_item.xml:32]: Why the name of this TextView is `some_id4` isn't meaningful

File: app/src/main/res/layout/card_view_item.xml [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            android:orientation="vertical">
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/indice_uv"
+                android:textAppearance="@style/indice_uv" />
+
+            <TextView
+                android:layout_width="wrap_content"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/card_view_item.xml:53]: In general, you need to have names for all those pieces of layout where you need to interact with. And names need to have a sense, so avoid tv1 text1 ecc.

File: app/src/main/res/layout/card_view_item.xml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_row="1"
+            android:layout_column="0"
+            android:orientation="vertical">
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:paddingTop="18dp"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/card_view_item.xml:69]: All the dimensions need to be in dimens file, not directly hardcoded into the layout.

File: app/src/main/res/layout/header_today_screen.xml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    <LinearLayout
+        android:id="@+id/home_screen_first_part"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        android:layout_marginBottom="54dp"
+        >
+
+    <TextView
+        android:id="@+id/palermo_sic"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/header_today_screen.xml:17]: Makes id more general and not specific.

File: app/src/main/res/layout/header_today_screen.xml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    android:layout_height="wrap_content"
+    android:id="@+id/headerLinearLayout"
+    android:orientation="vertical"
+    android:background="#F0EFFF">
+    <LinearLayout
+        android:id="@+id/home_screen_first_part"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        android:layout_marginBottom="54dp"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/header_today_screen.xml:13]: Avoid hardcoded dimens.

File: app/src/main/res/layout/header_today_screen.xml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        >
+        <TextView
+            android:layout_width="184dp"
+            android:layout_height="25dp"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="2dp"
+            android:text="Oggi,"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/header_today_screen.xml:36]: text here in Italian? I think that we need  to use strings, and set text programmatically.

File: app/src/main/res/layout/item_today_screen.xml [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+   >
+
+    <LinearLayout
+        android:layout_width="342dp"
+        android:layout_height="42dp"
+        android:layout_marginStart="24dp"
+        android:layout_marginEnd="40dp"
+        android:layout_marginTop="10dp"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/item_today_screen.xml:20]: All those hardcoded strings are wrong, and also wrong fixed width.

File: app/src/main/res/layout/item_today_screen.xml [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+            android:textAppearance="@style/some_id"
+
+            />
+
+        <ImageView
+            android:id="@+id/weatherImageView"
+            android:layout_width="24.61dp"
+            android:layout_height="24.42dp"
+            android:layout_marginTop="3.76dp"
+            android:layout_marginBottom="3.82dp"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/item_today_screen.xml:48]: ?? Displays render 2 pixels in 2 pixels so if you have .76 or .82 this dimension is rendered as 4dp. So avoid using those dimension with decimals.

File: app/src/main/res/layout/item_today_screen.xml [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+            android:layout_marginStart="52.75dp"
+            android:background="@drawable/humidity_image" />
+
+        <TextView
+            android:id="@+id/humidityTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="24dp"
+            android:layout_marginTop="4dp"
+            android:layout_marginBottom="4dp"
+            android:layout_marginStart="9.75dp"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/item_today_screen.xml:80]: Review dimensions

File: app/src/main/res/layout/today_screen.xml [hunk 13]
```diff
@@ -1,13 +1,71 @@
...
+
+        <LinearLayout
+            android:id="@+id/home_screen_first_part"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical">
+
+
+            <TextView
+                android:id="@+id/palermo_sic"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/today_screen.xml:25]: Why palermo_sic?

File: app/src/main/res/layout/today_screen.xml [hunk 14]
```diff
@@ -1,13 +1,71 @@
...
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="54dp"
+                android:orientation="vertical">
+
+                <TextView
+                    android:layout_width="184dp"
+                    android:layout_height="25dp"
+                    android:layout_marginStart="24dp"
+                    android:layout_marginTop="2dp"
+                    android:text="Oggi,"
```
Comment (ShAdOwNeT88) [inline app/src/main/res/layout/today_screen.xml:45]: Again hardcoded string.

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -1,5 +1,26 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+
+
+    <string name="oggi_domeni">Domenica 10 settembre</string>
```
Comment (ShAdOwNeT88) [inline app/src/main/res/values/strings.xml:5]: We do not need to have a date hardcoded like this, which is the purpose?

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -1,5 +1,26 @@
...
+    <string name="indice_uv">Indice UV</string>
+    <string name="some_id5">5/10</string>
+    <string name="umidit_">Umidità</string>
+    <string name="some_id6">60%</string>
+    <string name="vento">Vento</string>
+    <string name="sse_7km_h">SSE 7km/h</string>
+    <string name="copertura">Copertura</string>
+    <string name="some_id7">24%</string>
+    <string name="pioggia">Pioggia</string>
+    <string name="cm">0cm</string>
```
Comment (ShAdOwNeT88) [inline app/src/main/res/values/strings.xml:20]: In general, there are strings that are not necessary. And why are the names in Italian?

File: app/src/main/res/values/themes.xml [hunk 17]
```diff
@@ -13,4 +13,89 @@
         <item name="android:statusBarColor" >?attr/colorPrimaryVariant</item>
         <!-- Customize your theme here. -->
     </style>
+    <style name="some_id3">
```
Comment (ShAdOwNeT88) [inline app/src/main/res/values/themes.xml:16]: Why those styles are here? It is definitely wrong.

**Review by ShAdOwNeT88 — CHANGES_REQUESTED** [review]

File: app/src/main/java/co/develhope/meteoapp/MainActivity.kt (hunk 1 above)
Comment (Damidjo) [reply app/src/main/java/co/develhope/meteoapp/MainActivity.kt]: Done

File: app/src/main/java/co/develhope/meteoapp/TodayInfo.kt (hunk 2 above)
Comment (Damidjo) [reply app/src/main/java/co/develhope/meteoapp/TodayInfo.kt]: Done

**Pull request closed without merging**